source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -71,58 +118,258 @@ absl::optional<Arn> Filter::calculateRouteArn() {
     return absl::nullopt;
   }
 
-  return parseArn(settings->arn());
+  return *settings;
+}
+
+std::string Filter::resolveSettings() {
+  if (auto route_settings = getRouteSpecificSettings()) {
+    if (auto route_arn = parseArn(route_settings->arn())) {
+      arn_.swap(route_arn);
+      payload_passthrough_ = route_settings->payloadPassthrough();
+    } else {
+      ENVOY_LOG(warn, "Found route specific configuration but failed to parse Lambda ARN {}.",
```

Comment:
This can log at high rate in the data path. I would make this debug and add a stat.

Reply:
I'll change it to debug.
I plan to add many stats later in a separate PR. I'll add a TODO so I don't forget this one.